internal/terraform/context_apply_deferred_test.go

```diff
@@ -340,11 +344,59 @@ output "c" {
 			},
 		},
 	}
+
+	// resourceReadTest is a test that covers the behavior of reading resources
+	// in a refresh when the refresh is responding with a deferral.
+	resourceReadTest = deferredActionsTest{
+		configs: map[string]string{
+			"main.tf": `
+// TEMP: unknown for_each currently requires an experiment opt-in.
+// We should remove this block if the experiment gets stabilized.
+terraform {
+	experiments = [unknown_instances]
+}
+
+resource "test" "a" {
+	name = "deferred_read"
+}
+
+output "a" {
+	value = test.a
+}
+		`,
+		},
+		stages: []deferredActionsTestStage{
+			{
+				buildOpts: func(opts *PlanOpts) {
+					opts.Mode = plans.RefreshOnlyMode
```

Comment:
Could you add a second test in here for non refresh-only mode? There should be examples in some of my other PRs about running the test with some pre-existing state, then we want to make sure that if the read resource is deferred, it doesn't go ahead and try and plan the resource separately.